InMemoryFileSystemUnitTest/DirectoryTests.cs

```diff
@@ -1,43 +1,101 @@
-using Moq;
-using InMemoryFileSys;
+using NUnit.Framework;
+using System;
 
-namespace InMemoryFileSystemUnitTest
+namespace InMemoryFileSys.Tests
 {
+    [TestFixture]
     public class DirectoryTests
     {
-        private IDirectory _rootDirectory;
-        private Mock<IClock> _clockMock;
+        private Directory _rootDirectory;
 
         [SetUp]
         public void Setup()
         {
-            var mockClock = new Mock<IClock>();
-            mockClock.Setup(clock => clock.UtcNow).Returns(new DateTime(2024, 4, 1));
+            _rootDirectory = Directory.Root;
+        }
+
+        [Test]
+        public void AddFile_ValidRelativePath_FileAddedSuccessfully()
+        {
+            var relativePath = "test.txt";
+
+            var file = _rootDirectory.CreateFile(relativePath);
+
+            var path = file.Path;
+            Assert.IsNotNull(path);
+            Assert.IsTrue(path.EndsWith(relativePath));
+        }
+
+        [Test]
+        public void AddFile_NullOrWhiteSpaceRelativePath_ThrowsArgumentException()
+        {
+            string relativePath = null;
 
-            _rootDirectory = InMemoryFileSys.Directory.Root;
+            Assert.Throws<ArgumentException>(() => _rootDirectory.CreateFile(relativePath));
         }
-        
+
         [Test]
-        public void AddEntry_AddsFileToDirectory()
+        public void AddDirectory_ValidRelativePath_DirectoryAddedSuccessfully()
         {
-            var expectedFileName = "file1.txt";
 
-            _rootDirectory.AddEntry(expectedFileName, isFile: true);
+            var relativePath = "newDirectory";
+
+            var newDirectory = _rootDirectory.CreateDirectory(relativePath);
 
-            var path = _rootDirectory.GetPath(expectedFileName);
-            Assert.AreEqual("/" + expectedFileName, path);
+            var directoryPath = newDirectory.Path;
+            Assert.IsNotNull(directoryPath);
+            Assert.IsTrue(directoryPath.EndsWith(relativePath));
+            Assert.IsInstanceOf<Directory>(newDirectory);
```

Comment:
It is redundant check -- the compiler already did this for us as CreateDirectory returns a directory instance